src/tests.rs

```diff
@@ -106,6 +106,89 @@ macro_rules! nodes {
     }};
 }
 
+#[derive(Debug)]
+enum MergeStateTag {
```

Comment:
This exists for two reasons:

* For tests, the `{Local, Remote}Only` vs. `{Local, Remote}` distinction doesn't matter.
* Building up `MergedNode`s is kind of a pain, since we need to pull in nodes from `local_tree` and `remote_tree`. So instead we use flat structs and data-less enums, and have a `PartialEq` implementation that compares with `MergedNode` below.

Reply:
ditto - seems worth making an actual code comment?